### PR TITLE
Add testing and typing to analysis.py

### PR DESCRIPTION
**Pull request by henryre**
See title. Primarily adding unit tests for analysis.py
Also:
* Handling of NaN outputs in `lf_empirical_accuracies`
* Trim poorly defined columns in `single_lf_summary`
* Replace `issparse` for type-ability
* Removed that goofball `global` since it's unused

**Test plan**
Added tests...obviously

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -31,7 +31,7 @@ commands = mypy -p snorkel
 description = run coverage checks
 basepython = python3
 usedevelop = True
-commands = pytest --cov=snorkel test/
+commands = python -m pytest --cov=snorkel test/
```
Comment (bhancock8) [inline tox.ini]: Why was this necessary to add?

File: test/labeling/test_analysis.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    def test_label_overlap(self) -> None:
+        self.assertEqual(label_overlap(self.L), 4 / 6)
+
+    def test_label_conflict(self) -> None:
+        self.assertEqual(label_conflict(self.L), 3 / 6)
+
+    def test_lf_polarities(self) -> None:
+        polarities = lf_polarities(self.L)
+        self.assertEqual(polarities, [[2, 3], [], [1, 3], 3, [1, 2], 1])
```
Comment (bhancock8) [inline test/labeling/test_analysis.py]: lf_polarities should return a consistent type--probably always a list?

File: tox.ini (hunk 1 above)
Comment (henryre) [reply tox.ini]: It wasn't strictly but just standardized the pytest call (see the py env above) when debugging the PyArrow+PyTorch issue

File: test/labeling/test_analysis.py (hunk 2 above)
Comment (henryre) [reply test/labeling/test_analysis.py]: https://github.com/HazyResearch/snorkel/pull/1168/files#diff-5839d69014187bdf1a2f60ccee964363R73

Comment (bhancock8) [reply test/labeling/test_analysis.py]: Right, I see that the code _allows_ for both right now. But I think we should change it so that it only returns a List of Lists.

Comment (henryre) [reply test/labeling/test_analysis.py]: Ah I see, you're saying the lone ints are unexpected. I agree. Can change now.

Comment (henryre) [reply test/labeling/test_analysis.py]: Or add a flag?

Comment (bhancock8) [reply test/labeling/test_analysis.py]: The flag would only be applicable some of the time, since multi-polar LFs would still need to be a collection of some sort. I say just make it always a list of lists.

Comment (henryre) [reply test/labeling/test_analysis.py]: Cool on it

Comment (henryre) [reply test/labeling/test_analysis.py]: Will do in a stacked PR so as not to complicate rebases. Not really in scope for this diff either.
